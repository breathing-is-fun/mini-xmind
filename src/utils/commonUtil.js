export function noop() {}

export const isSameCoordinate = (nextProps, nextState, currentKey) => {
  if (!currentKey) {
    return false;
  }
  const currentProps = nextProps.data[currentKey];
  const currentState = nextState.data[currentKey];

  if (currentProps.x != currentState.x || currentProps.y != currentState.y) {
    return true;
  }

  return false;
};
