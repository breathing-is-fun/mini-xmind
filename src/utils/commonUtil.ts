import { BlockGroupProps, BlockGroupState } from '../tools/BlockGroup';
import { TagGroupProps, TagGroupState } from '../tools/TagGroup';

export function noop() {}

export const isSameCoordinate = (
  nextProps: BlockGroupProps | TagGroupProps,
  nextState: BlockGroupState | TagGroupState,
  currentKey?: string,
) => {
  if (!currentKey) {
    return false;
  }
  const currentProps = nextProps.data[currentKey];
  const currentState = nextState.data[currentKey];

  if (currentProps.x != currentState.x || currentProps.y != currentState.y) {
    return true;
  }

  return false;
};
