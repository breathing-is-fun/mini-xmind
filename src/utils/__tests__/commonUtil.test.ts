import { isSameCoordinate } from '../commonUtil';

describe('commonUtil', () => {
  it('when currentKey is null, return', () => {
    expect(isSameCoordinate({}, {}, null)).toBe(false);
    expect(
      isSameCoordinate(
        {
          data: {
            block: { x: 10, y: 10 },
          },
        },
        {
          data: {
            block: { x: 10, y: 10 },
          },
        },
        'block',
      ),
    ).toBe(false);
    expect(
      isSameCoordinate(
        {
          data: {
            block: { x: 11, y: 10 },
          },
        },
        {
          data: {
            block: { x: 10, y: 10 },
          },
        },
        'block',
      ),
    ).toBe(true);
  });
});
