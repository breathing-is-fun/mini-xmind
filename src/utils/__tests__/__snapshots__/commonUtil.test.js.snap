// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`go should call ReactDOM.render 1`] = `[Function]`;
